src/profile/hooks/useProfileMutations.ts

```diff
@@ -0,0 +1,145 @@
+import {
+  useMutation,
+  FetchResult,
+  Reference,
+  ApolloCache,
+  ApolloError,
+} from '@apollo/client';
+import { loader } from 'graphql.macro';
+import { useContext } from 'react';
+import { Modifiers } from '@apollo/client/cache/core/types/common';
+
+import { UpdateMyProfileVariables } from '../../graphql/generatedTypes';
+import {
+  ProfileRoot,
+  UpdateProfileRoot,
+  UpdateProfileData,
+  AnyObject,
+} from '../../graphql/typings';
+import {
+  FormValues,
+  EditDataType,
+  basicDataType,
+  additionalInformationType,
+} from '../helpers/editData';
+import { updateMutationVariables } from '../helpers/updateMutationVariables';
+import { ProfileContext, ProfileContextData } from '../context/ProfileContext';
+
+const UPDATE_PROFILE = loader('../../profile/graphql/UpdateMyProfile.graphql');
+const MY_PROFILE = loader('../../profile/graphql/MyProfileQuery.graphql');
+
+export type QueryResult = FetchResult<ProfileRoot, AnyObject, AnyObject>;
+export type UpdateResult = FetchResult<UpdateProfileRoot>;
+
+type UpdateProfileFunction = (
+  formValues: Partial<FormValues>,
+  profile?: ProfileRoot
+) => Promise<UpdateResult>;
+
+export type MutationReturnType = {
+  error: ApolloError | undefined;
+  loading: boolean;
+  profileData?: ProfileRoot;
+  update: UpdateProfileFunction;
+};
+
+type UpdatableProfileDataKeys = keyof Omit<
+  UpdateProfileData,
+  '__typename' | 'id'
+>;
+
+export type ValuesOfUpdateProfileData = UpdateProfileData[UpdatableProfileDataKeys];
```

Comment:
Did not find any usages of this type outside of this file, so maybe the export is unnecessary?

Reply:
Removed.